.github/CONTRIBUTING.md

```diff
@@ -320,12 +300,60 @@ def test_explain_what_is_being_tested(tmpdir):
         ...
     )
     trainer.fit(model)
-    result = trainer.test()
+    trainer.test() # [OPTIONAL]
 
     # assert the behaviour is correct.
     assert ...
-    assert ...
 ```
+run our/your test with
+```bash
+python -m pytest tests/..../...py::test_explain_what_is_being_tested --verbose --capture=no
+```
+
+#### How to contribute bugfix/feature?
+
+At this moment we have separate streams/branches for bugfixes and features as we utilize releasing from default branch (`master`).
+Bugfix shall land in this `master` branch compare to feature which shall land in `release/1.2-dev`.
+This means that in this time you starting your contribution and creating branch according to question 2) you shall start this new branch from master or future release.
+Later in PR creation also pay attention to set properly the target branch, a simple understanding is that the staring (base) and target branch shall be the same.
+
+_Note, that this flow may change after 1.2 release as we will adjust releasing strategy._
+
+#### How to fix PR with mixed base nad target branch?
+
+It can easily happen that you start your PR as bug-fix (and already created a PR accordingly) which turned to  be rather a feature or other way around.
+Do not panic, the solution is very straight forward and a quite simple.
+You need to do this two steps and in fact it do not matter much on the order:
+   - Ask someone from Core to change the base/target branch to the correct one
+   - Rebase or cherry-pick your commit onto correct base branch...
+     
+Let's show how to dela with the git...
+the sample case is moving PR from `master` to `release/1.2-dev` assuming my branch name is `my-branch`
+and the last true master commit is `ccc111` and your first commit is `mmm222`.
+   * **Cherry-picking** way
+     ```bash
+     git checkout my-branch
+     # create local backup of your branch
+     git checkout -b my-branch-backup
+     # reset your branch to the correct base
+     git reset release/1.2-dev --hard
+     # ACTION: this step is much easier to do with IDE
+     #  so open one and cherry pich you last commits from `my-branch-backup`
+     #  resolve all eventual conflict as the new base may contain different code
+     # when all done, push back to the open PR
+     git push -f 
+     ```
+   * **Rebasing way**, see more about [rebase onto usage](https://womanonrails.com/git-rebase-onto)
+     ```bash
+     git checkout my-branch
+     # rebase your commits on the correct branch
+     git rebase --onto release/1.2-dev ccc111
+     # if there are no collision you shall see just success
+     #  eventually you would need to resolve collision and in such case follow instruction in terminal
+     # when all done, push back to the open PR
+     git push -f
```

Comment:
```suggestion
It can easily happen that you start your PR as a bug-fix (and already created a PR accordingly) which turned to be rather a feature or the other way around.
Do not panic, the solution is very straightforward and quite simple.
You need to do these two steps and in fact, it does not matter much on the order:
   - Ask someone from Core to change the base/target branch to the correct one
   - Rebase or cherry-pick your commit onto the correct base branch...
     
Let's show how to deal with the git...
the sample case is moving PR from `master` to `release/1.2-dev` assuming my branch name is `my-branch`
and the last true master commit is `ccc111` and your first commit is `mmm222`.
   * **Cherry-picking** way
     ```bash
     git checkout my-branch
     # create a local backup of your branch
     git checkout -b my-branch-backup
     # reset your branch to the correct base
     git reset release/1.2-dev --hard
     # ACTION: this step is much easier to do with IDE
     #  so open one and cherry-pick your last commits from `my-branch-backup`
     #  resolve all eventual conflict as the new base may contain different code
     # when all done, push back to the open PR
     git push -f 
     ```
   * **Rebasing way**, see more about [rebase onto usage](https://womanonrails.com/git-rebase-onto)
     ```bash
     git checkout my-branch
     # rebase your commits on the correct branch
     git rebase --onto release/1.2-dev ccc111
     # if there is no collision you shall see just success
     #  eventually you would need to resolve collision and in such case follow the instruction in terminal
     # when all done, push back to the open PR
     git push -f
```